File: mobile/lib/features/workouts/models/workout_exercise_completion_response.dart
class WorkoutExerciseCompletionResponse {
  const WorkoutExerciseCompletionResponse({
    required this.workoutDayId,
    required this.workoutExerciseId,
    required this.isCompleted,
    required this.completedExercises,
    required this.totalExercises,
  });

  final String workoutDayId;
  final String workoutExerciseId;
  final bool isCompleted;
  final int completedExercises;
  final int totalExercises;

  factory WorkoutExerciseCompletionResponse.fromJson(
      Map<String, dynamic> json,
      ) {
    return WorkoutExerciseCompletionResponse(
      workoutDayId: json['workoutDayId'] as String,
      workoutExerciseId: json['workoutExerciseId'] as String,
      isCompleted: json['isCompleted'] as bool,
      completedExercises: (json['completedExercises'] as num).toInt(),
      totalExercises: (json['totalExercises'] as num).toInt(),
    );
  }
}

File: src/GymFlow.Application/DTOs/Workouts/WorkoutExerciseCompletionResponse.cs
namespace GymFlow.Application.DTOs.Workouts;

public class WorkoutExerciseCompletionResponse
{
    public Guid WorkoutDayId { get; set; }

    public Guid WorkoutExerciseId { get; set; }

    public bool IsCompleted { get; set; }

    public int CompletedExercises { get; set; }

    public int TotalExercises { get; set; }
}